be/src/exprs/str_to_map.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <stack>
+
+#include "column/array_column.h"
+#include "column/binary_column.h"
+#include "column/column_helper.h"
+#include "column/column_viewer.h"
+#include "column/map_column.h"
+#include "exprs/string_functions.h"
+#include "gutil/strings/split.h"
+#include "util/utf8.h"
+
+namespace starrocks {
+
+/**
+* @param: [array_string, delimiter]
+* @paramType: [ArrayBinaryColumn, BinaryColumn]
+* @return: MapColumn map<string,string>
+
+ the original str_to_map(str, del1, del2) is rewritten to str_to_map(split(str, del1), del2), the first input results
```

Comment:
1. What's this function used for?
2. Why need two delimiters?
3. Could the delimiters be mulitple characters?

Reply:
what if extra space in the string, like `{1:2,      2:3}` ?

Reply:
more msg please refer to doc.
the extra space will be in map, as they are token into string.